### PR TITLE
Angle helper

### PR DESCRIPTION
**Pull request by Hamms**
For helping kids visualize angles!

Comes in two versions:

Dropdown Mode

dropdown

And text input mode:

text

Also supports both "left" and "right" modes:

leftright

Next step is to update our existing artist blocks to use these fields (as in the gifs) and add eyes tests.

### CODE REVIEW

**Comment by bcjordan** [comment]
Might be worth taking a peek at:

https://google.github.io/closure-library/api/goog.math.Vec2.html
https://google.github.io/closure-library/api/goog.math.Coordinate.html
https://github.com/google/closure-library/blob/master/closure/goog/math/vec2.js#L196

in case they do any of the lifting the helpers are doing.

File: core/ui/fields/angle_helper.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    'stroke-width': this.strokeWidth,
+  }, this.svg);
+
+  // Draw markers around the edge.
+  for (var a = 15; a < 360; a += 15) {
+    Blockly.createSvgElement('line', {
+      'stroke-linecap': 'round',
+      'x1': this.center.x + this.lineLength,
+      'y1': this.center.y,
+      'x2': this.center.x + this.lineLength - (a % 90 == 0 ? 15 : a % 45 == 0 ? 10 : 5),
```
Comment (bcjordan) [inline core/ui/fields/angle_helper.js]: Could maybe use a variable/constant/helper extraction here for clarity

File: core/ui/fields/angle_helper.js [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  this.circleCenter = Blockly.AngleHelper.polarToCartesian(
+    this.center.x,
+    this.center.y,
+    this.lineLength,
+    this.turnRight ? this.angle : -this.angle
+  );
+
+  this.arc;
+  this.circle;
+  this.svg;
```
Comment (bcjordan) [inline core/ui/fields/angle_helper.js]: Do we mostly skip the `_` suffix naming convention for private properties but keep it for private methods?

File: core/ui/fields/angle_helper.js [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+  this.lineLength = (circumference / 2) - this.circleR - this.strokeWidth;
+
+  this.circleCenter = Blockly.AngleHelper.polarToCartesian(
+    this.center.x,
+    this.center.y,
+    this.lineLength,
+    this.turnRight ? this.angle : -this.angle
+  );
+
+  this.arc;
```
Comment (bcjordan) [inline core/ui/fields/angle_helper.js]: (Though I'm not necessarily against this convention as long as it doesn't upset any linters)

File: tests/angle_helper_test.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  });
+}
+
+function test_describeArc() {
+  function evaluate(startAngle, endAngle, expected) {
+    var actual = Blockly.AngleHelper.describeArc(0, 0, 100, startAngle, endAngle);
+    assertEquals(expected, actual);
+  }
+
+
```
Comment (bcjordan) [inline tests/angle_helper_test.js]: (Nit: extra line)

**Comment by Hamms** [comment]
@bcjordan PTAL!

**Comment by bcjordan** [comment]
`goog.math.Awesome`!! Very nice, LGTM! 🎉 👍

**Pull request merged**